src/samples/dolphin/dolphin.cpp

```diff
@@ -151,11 +186,32 @@ void Dolphin::onUpdate()
         {
             printf("Pickup %d Dist %d\n", i, dist);
             oysters[i].disappear();
+            player.setLife(player.getLifes() + 1);
             engine->audio.setADPCMVolume(30, 3);
             engine->audio.playADPCM(pickupSound, 3);
         }
     }
 
+    for (int i = 0; i < MINES_COUNT; i++)
+    {
+        mines[i].update();
+        Vector3 vecDist = mines[i].mesh.position - player.mesh.position;
+        float dist = Math::sqrt((vecDist.x * vecDist.x) +
+                                (vecDist.y * vecDist.y) +
+                                (vecDist.z * vecDist.z));
+        if (dist < 20.F && !mines[i].getExplosionTicks())
+        {
+            printf("Vecdist %f,%f,%f\n", vecDist.x, vecDist.y, vecDist.z);
```

Comment:
I see that we have a lot of different logic there (in onUpdate()). Please consider in future, to move logic to another classes.